paperqa/agents/__init__.py

```diff
@@ -0,0 +1,500 @@
+
+import ast
+import logging
+import operator
+import os
+import shutil
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import yaml
+from typing_extensions import Annotated
+
+from .. import __version__
+from ..utils import get_loop, pqa_directory
+
+try:
+    import anyio
+    import typer
+    from rich.console import Console
+    from rich.logging import RichHandler
+
+    from .main import aagent_query, asearch, configure_agent_logging
+    from .models import AnswerResponse, MismatchedModelsError, QueryRequest
+    from .search import SearchIndex, get_directory_index
+
+except ImportError as e:
+    raise ImportError(
+        '"agents" module is not installed please install it using "pip install paper-qa[agents]"'
+    ) from e
+
+app = typer.Typer()
+
+logging.basicConfig(
+    level="INFO",
+    format="%(message)s",
+    datefmt="[%X]",
+    handlers=[
+        RichHandler(
+            rich_tracebacks=True,
+            markup=True,
+            show_path=False,
+            show_level=False,
+            console=Console(force_terminal=True),
+        )
+    ],
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_file_times(path: os.PathLike | str) -> dict[str, str]:
+    # Get the stats for the file/directory
+    stats = os.stat(path)
+
+    # Get created time (ctime)
+    created_time = datetime.fromtimestamp(stats.st_ctime)
+
+    # Get modified time (mtime)
+    modified_time = datetime.fromtimestamp(stats.st_mtime)
+
+    return {
+        "created_at": created_time.strftime("%Y-%m-%d %H:%M:%S"),
+        "modified_at": modified_time.strftime("%Y-%m-%d %H:%M:%S"),
+    }
+
+
+def parse_dot_to_dict(str_w_dots: str, value: str) -> dict[str, Any]:
+    parsed: dict[str, Any] = {}
+    for key in str_w_dots.split(".")[::-1]:
+        if not parsed:
+            try:
+                eval_value = ast.literal_eval(value)
+                if isinstance(eval_value, (set, list)):
+                    parsed[key] = eval_value
+                else:
+                    parsed[key] = value
+            except (ValueError, SyntaxError):
+                parsed[key] = value
+        else:
+            parsed = {key: parsed}
+    return parsed
+
+
+def pop_nested_dict_recursive(d: dict[str, Any], path: str) -> tuple[Any, bool]:
+    """
+    Pop a value from a nested dictionary using a period-separated path.
+
+    Recursively remove empty dictionaries after popping.
+    """
+    keys = path.split(".")
+
+    if len(keys) == 1:
+        if keys[0] not in d:
+            raise KeyError(f"Key not found: {keys[0]}")
+        value = d.pop(keys[0])
+        return value, len(d) == 0
+
+    if keys[0] not in d or not isinstance(d[keys[0]], dict):
+        raise KeyError(f"Invalid path: {path}")
+
+    value, should_remove = pop_nested_dict_recursive(d[keys[0]], ".".join(keys[1:]))
+
+    if should_remove:
+        d.pop(keys[0])
+
+    return value, len(d) == 0
+
+
+def get_current_settings(
```

Comment:
```suggestion
def get_settings(
```

No need for the word "current" imo